CDCgov/prime-reportstream
# Updates to fhirpath exception handling

This PR fixes some oddities within the error handling surrounding our fhirpath utils in ReportStream.

Test Steps:
1. You'll need to run prime commands to test this. Use your favorite command to build the project, or use mine: (from root directory) `./gradlew fatjar`
2. Next bring up the fhirpath evaluator. This is not the only way to get at these code changes, just the quickest. (from the prime-router directory) `./prime fhirpath --input-file="src/test/resources/fhirengine/engine/bundle_multiple_observations.fhir"`
3. Happy Path
   1. `Bundle.entry.resource.ofType(Observation).where(code.coding.code = '94558-5')`
   2. Expect results from the bundle to be returned.
4. Invalid Syntax
   1. `Bundle.entry.resource.ofType(Observation).where(component.where(code.coding.code = '94558-5').exists()`
   2. Expect a FHIRLexerException to be logged out, but the util does not unexpectedly exit.
5. "The Extension() Bug"
   1. `Bundle.extension(blah).value`
   2. Expect a IndexOutOfBoundsException to be logged out, but the util does not unexpected exit.
6. Invalid Resource Type
   1. `Bundle.entry.resource.ofType(Messi)`
   2. Expect an unhandled PathEngineExcept, and for the util to exit.
   3. I did not add this to the handled exceptions, but could be convinced to. 

## Changes
- Updated FhirPathUtils.evaluate() to only catch expected errors (Invalid fhirpath syntax, and "the extension() bug"
- Error handling related to "the extension() bug" was due to fhirpath expressions containing non-string values provided for an extension. Ex: `Bundle.extension('blah')` vs `Bundle.extension(blah)`. This is now being specifically handled by catching the resulting `IndexOutOfBoundsException`. 
- Removed several instances of unnecessary error handling
  - ProcessFhirCommands::FhirPathCommand.Run()
    - FhirPathCommand.setFocusResource() and FhirPathCommand.evaluatePath called in Run() are both properly catching exceptions.
  - ProcessFhirCommands::FhirPathCommand.evaluatePath()
    - Method was try/catching exceptions around FhirPathUtils.evaluate() but all exceptions are handled within that method.
- Updated incorrect error messaging.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #14540
